Branch-SDK/src/main/java/io/branch/referral/Branch.java

```diff
@@ -2414,6 +2414,9 @@ else if (isReInitializing) {
                 expectDelayedSessionInitialization(true);
             }
 
+            AttributionReportingManager attributionManager = AttributionReportingManager.INSTANCE;
+            attributionManager.checkMeasurementApiStatus(branch.context_);
```

Comment:
What's the overhead of this set up? Does this need to be part of the main execution or can it (should it) be backgrounded?

Reply:
In the checkMeasurementApiStatus method, the getMeasurementApiStatus() call is done on an executor. Is this the right approach?